src/ai/susi/server/api/aaa/ChangeUserRoles.java

```diff
@@ -0,0 +1,112 @@
+package ai.susi.server.api.aaa;
+
+import ai.susi.DAO;
+import ai.susi.SusiServer;
+import ai.susi.json.JsonObjectWithDefault;
+import ai.susi.server.*;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import javax.servlet.Servlet;
+import javax.servlet.http.HttpServletResponse;
+import java.lang.reflect.Constructor;
+
+/**
+ * Created by chetankaushik on 30/05/17.
+ * This is a servlet that is used to change the role of any user
+ * This requires Admin login
+ * http://127.0.0.1:4000/aaa/changeRoles this accepts two post parameters-
+ *  1. user    --> The username of the user whose role is to be modified
+ *  2. role    --> the new role of the user
+ */
+
+public class ChangeUserRoles extends AbstractAPIHandler implements APIHandler {
+    private static final long serialVersionUID = -1432553481906185711L;
+
+    @Override
+    public BaseUserRole getMinimalBaseUserRole() { return BaseUserRole.ADMIN; }
+
+    @Override
+    public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
+        return null;
+    }
+
+    @Override
+    public String getAPIPath() {
+        return "/aaa/changeRoles";
+    }
+
+
+    @Override
+    public JSONObject serviceImpl(Query query, HttpServletResponse response, Authorization authorization, final JsonObjectWithDefault permissions) throws APIException {
+
+        JSONObject result = new JSONObject();
+        if (authorization.getBaseUserRole().toString().equals("ADMIN")) {
```

Comment:
@Orbiter Is this check necessary? As I understood you, it's enough to override `getMinimalBaseUserRole()`.